grid/app/main/auth/federated.py

```diff
@@ -0,0 +1,81 @@
+# PyGrid imports
+from ..codes import MSG_FIELD, RESPONSE_MSG, CYCLE, FL_EVENTS
+from ..processes import process_manager
+
+# Generic imports
+import jwt
+import uuid
+import json
+import base64
+import requests
+
+
+def verify_token(auth_token, model_name):
+    server_config, _ = process_manager.get_configs(name=model_name)
+    auth_config = server_config.get("authentication", {})
+
+    """stub DB vars"""
+    JWT_VERIFY_API = auth_config.get("endpoint", None)
+    RSA = auth_config.get("JWT_with_RSA", None)
+
+    if RSA:
+        pub_key = auth_config.get("pub_key", None)
+    else:
+        SECRET = auth_config.get("secret", "very long a$$ very secret key phrase")
+    """end stub DB vars"""
+
+    HIGH_SECURITY_RISK_NO_AUTH_FLOW = False if JWT_VERIFY_API is not None else True
```

Comment:
Hmm, I think "no auth" mode should be when there's no "authentication" section at all?
Not when just `endpoint` is missing?